Objective-C/Objective-C.sublime-syntax

```diff
@@ -13,200 +15,334 @@ first_line_match: |-
-          scope: punctuation.definition.string.end.objc
-          pop: true
-        - match: \b(?i:AND|OR|NOT|IN)\b
-          scope: keyword.operator.logical.predicate.cocoa.objc
-        - match: \b(?i:ALL|ANY|SOME|NONE)\b
-          scope: constant.language.predicate.cocoa.objc
-        - match: \b(?i:NULL|NIL|SELF|TRUE|YES|FALSE|NO|FIRST|LAST|SIZE)\b
-          scope: constant.language.predicate.cocoa.objc
-        - match: \b(?i:MATCHES|CONTAINS|BEGINSWITH|ENDSWITH|BETWEEN|LIKE)\b
-          scope: keyword.operator.comparison.predicate.cocoa.objc
-        - match: \b(?i:C(ASEINSENSITIVE|I))\b
-          scope: keyword.other.modifier.predicate.cocoa.objc
-        - match: \b(?i:ANYKEY|SUBQUERY|CAST|TRUEPREDICATE|FALSEPREDICATE)\b
-          scope: keyword.other.predicate.cocoa.objc
-        - match: '\\(\\|[abefnrtv''"?]|[0-3]\d{,2}|[4-7]\d?|x[a-zA-Z0-9]+)'
-          scope: constant.character.escape.objc
-        - match: \\.
-          scope: invalid.illegal.unknown-escape.objc
+        1: keyword.operator.accessor.objc
+        2: variable.other.readwrite.member.objc
+    - include: pop-immediately
 
-  unique-strings:
-    - match: '@"'
+  preprocessor:
+    - meta_prepend: true
+    - match: '{{dir_start}}(import)\b'
+      scope: keyword.control.import.objc
+      push: preprocessor.include
+
+  string:
+    - meta_prepend: true
+    - match: \@\"
       scope: punctuation.definition.string.begin.objc
-      push:
-        - meta_scope: string.quoted.double.objc
-        - match: '"'
-          scope: punctuation.definition.string.end.objc
-          pop: true
-        - include: scope:source.c#string_escaped_char
-        - match: '%@'
-          scope: constant.other.placeholder.objc
-        - include: scope:source.c#string_placeholder
+      push: string.double-ns
+    - match: \@\'
+      scope: punctuation.definition.string.begin.objc
+      push: string.single
+  string.double-ns:
+    - meta_scope: string.quoted.double.objc
+    - include: string.double
+    - match: '%@'
+      scope: constant.other.placeholder.objc
+
+  available:
+    - match: \b(iOS|macOS|watchOS|tvOS|visionOS)\b
+      scope: support.constant.platform.objc
+    - match: \b(\d[\d\.]+)\b
+      captures:
+        1: constant.numeric.version.objc
+    - include: expression
+    - match: \)
+      scope: punctuation.section.group.end.objc
+      pop: 1
 
-  unique-keywords:
-    - match: (@)(dynamic|synthesize)\b
-      scope: keyword.other.property.directive.objc
-      captures:
-        1: punctuation.definition.keyword.objc
-    - match: \b(NS_DURING|NS_HANDLER|NS_ENDHANDLER)\b
-      scope: keyword.control.macro.objc
-    - match: (@)(try|catch|finally|throw)\b
-      scope: keyword.control.exception.objc
+  selector-obj:
+    - meta_scope: meta.selector.objc
+    - meta_content_scope: meta.selector.method-name.objc
+    - match: \)
+      scope: punctuation.section.group.end.objc
+      pop: 1
+    - match: .
+      scope: support.function.objc
```

Comment:
A `.` pattern causes each character to become a single token, which is not ideal for various reasons and should be avoided if possible.